text/0229-deprecate-testing-restricted-resolver.md

```diff
@@ -0,0 +1,217 @@
+  * `setupModelTest`
+
+### Non Component Test APIs
+
+The migration path for `moduleFor`, `moduleForModel`, `setupTest`, and 
+`setupModelTest` is very simple:
+
+```js
+// ember-qunit
+
+// before
+moduleFor('service:session');
+
+moduleFor('service:session', {
+  unit: true
+});
+
+moduleFor('service:session', {
+  needs: ['type:thing']
+});
+
+// after
+moduleFor('service:session', {
+  integration: true
+});
+```
+
+```js
+// ember-mocha
+
+// before
+describe('Session Service', function() {
+  setupTest('service:session');
+  
+  // ...snip...
+});
+
+describe('Session Service', function() {
+  setupTest('service:session', { unit: true });
+  
+  // ...snip...
+});
+
+describe('Session Service', function() {
+  setupTest('service:session', { needs: [] });
+  
+  // ...snip...
+});
+
+// after
+
+describe('Session Service', function() {
+  setupTest('service:session', { integration: true });
+  
+  // ...snip...
+});
+```
+
+The main change is adding `integration: true` to options (and removing `unit` or `needs`
+if present).
+
+### Component Test APIs
+
+Implicitly relying on "unit test mode" has been deprecated for quite some time 
+([introduced 2015-04-07](https://github.com/emberjs/ember-test-helpers/pull/38)),
+so all consumers of `moduleForComponent` and `setupComponentTest` are specifying
+one of the "test type options" (`unit`, `needs`, or `integration`).
+
+This RFC proposes to deprecate completely using `unit` or `needs` options with
+`moduleForComponent` and `setupComponentTest`. The vast majority of component tests
+should be testing via `moduleForComponent` / `setupComponentTest` with the `integration: true`
+option set, but on some rare occaisions it is easier to use the "unit test" style is
+desired (e.g. non-rendering test) these tests should be migrated to using `moduleFor` 
+/ `setupTest` directly.
+
+```js
+// ember-qunit
+
+// before
+moduleForComponent('display-page', {
+  unit: true
+});
+
+moduleForComponent('display-page', {
+  needs: ['type:thing']
+});
+
+// after
+
+moduleFor('component:display-page', {
+  integration: true
+});
+```
+
+```js
+// ember-mocha
+describe('DisplayPageComponent', function() {
+  setupComponentTest('display-page', { unit: true });
+  
+  // ...snip...
+});
+
+describe('DisplayPageComponent', function() {
+  setupComponentTest('display-page', { needs: [] });
+  
+  // ...snip...
+});
+
+// after
+
+describe('DisplayPageComponent', function() {
+  setupTest('component:display-page', { integration: true });
+  
+  // ...snip...
+});
+```
+
+## Ecosystem Updates
+
+The blueprints in all official projects (and any provided by popular
+addons) will need to be updated to avoid triggering a deprecation.
+
+This includes:
+
+* `ember-source`
+* `ember-data`
+* `ember-cli-legacy-blueprints`
+* Others?
+
+## Remove Deprecated `unit` / `needs` Options
+
+Once the changes from this RFC are made, we will be able to remove
+support for `unit` and `needs` options from `ember-test-helpers`,
```

Comment:
I definitely understand wanting to avoid the churn, but it feels incorrect to me to leave `integration: true` sitting in a bunch of tests with no apparent reason for it being there. One concern about this is that even though the flag will no longer do anything, developers may still believe it has some effect on the behavior of the tests.

Maybe rather than introduce a deprecation that logs on every single test that has the flag, we log a message the first time it is encountered? Just to ensure context is given.